src/System.Net.NameResolution/src/System/Net/DNS.cs

```diff
@@ -0,0 +1,383 @@
+            IAsyncResult asyncResult = HostResolutionBeginHelper(hostNameOrAddress, false, requestCallback, stateObject);
+
+            if (Logging.On) Logging.Exit(Logging.Sockets, "DNS", "BeginGetHostEntry", asyncResult);
+            return asyncResult;
+        } // BeginResolve
+
+        public static IAsyncResult BeginGetHostEntry(IPAddress address, AsyncCallback requestCallback, object stateObject)
+        {
+            if (Logging.On) Logging.Enter(Logging.Sockets, "DNS", "BeginGetHostEntry", address);
+            IAsyncResult asyncResult = HostResolutionBeginHelper(address, true, true, requestCallback, stateObject);
+
+            if (Logging.On) Logging.Exit(Logging.Sockets, "DNS", "BeginGetHostEntry", asyncResult);
+            return asyncResult;
+        } // BeginResolve
+
+        public static IPHostEntry EndGetHostEntry(IAsyncResult asyncResult)
+        {
+            if (Logging.On) Logging.Enter(Logging.Sockets, "DNS", "EndGetHostEntry", asyncResult);
+            IPHostEntry ipHostEntry = HostResolutionEndHelper(asyncResult);
+
+            if (Logging.On) Logging.Exit(Logging.Sockets, "DNS", "EndGetHostEntry", ipHostEntry);
+            return ipHostEntry;
+        } // EndResolve()
+
+        public static IAsyncResult BeginGetHostAddresses(string hostNameOrAddress, AsyncCallback requestCallback, object state)
+        {
+            if (Logging.On) Logging.Enter(Logging.Sockets, "DNS", "BeginGetHostAddresses", hostNameOrAddress);
+            IAsyncResult asyncResult = HostResolutionBeginHelper(hostNameOrAddress, true, requestCallback, state);
+
+            if (Logging.On) Logging.Exit(Logging.Sockets, "DNS", "BeginGetHostAddresses", asyncResult);
+            return asyncResult;
+        } // BeginResolve
+
+        public static IPAddress[] EndGetHostAddresses(IAsyncResult asyncResult)
+        {
+            if (Logging.On) Logging.Enter(Logging.Sockets, "DNS", "EndGetHostAddresses", asyncResult);
+            IPHostEntry ipHostEntry = HostResolutionEndHelper(asyncResult);
+
+            if (Logging.On) Logging.Exit(Logging.Sockets, "DNS", "EndGetHostAddresses", ipHostEntry);
+            return ipHostEntry.AddressList;
+        } // EndResolveToAddresses
+
+        //************* Task-based async public methods *************************
+        public static Task<IPAddress[]> GetHostAddressesAsync(string hostNameOrAddress)
+        {
+            return Task<IPAddress[]>.Factory.FromAsync(BeginGetHostAddresses, EndGetHostAddresses, hostNameOrAddress, null);
+        }
+
+        public static Task<IPHostEntry> GetHostEntryAsync(IPAddress address)
+        {
+            return Task<IPHostEntry>.Factory.FromAsync(BeginGetHostEntry, EndGetHostEntry, address, null);
+        }
+
+        public static Task<IPHostEntry> GetHostEntryAsync(string hostNameOrAddress)
+        {
+            return Task<IPHostEntry>.Factory.FromAsync(BeginGetHostEntry, EndGetHostEntry, hostNameOrAddress, null);
+        }
```

Comment:
Done.

Please open an issue to track this here as well as Desktop (/cc: @SidharthNabar). 

Given the simple transformation required, I think this could be optimized at the compiler-side. It would be great if you could open an issue for https://github.com/dotnet/roslyn.